search/search_service/proxy/es_search_proxy.py

```diff
@@ -317,3 +336,101 @@ def search(self, *,
                                                    resource_types=resource_types)
 
         return formatted_response
+
+    def get_document_by_key(self,
+                            resource_key: str,
+                            resource_type: Resource,
+                            field: str) -> Document:
+        key_query = {
+            resource_type: Q(TERM_QUERY, key=resource_key),
+        }
+        response: Response = self.execute_queries(queries=key_query,
+                                                  page_index=0,
+                                                  results_per_page=1)[0]
+        if response.success():
+            results_count = response.hits.total.value
+            if results_count > 0:
+                es_result = response.hits.hits[0]
+                resource_es_id = es_result._id
+
+                # return document
+                resource_str = resource_type.name.lower()
+                resource_index = f"{resource_str}_search_index"
+                return RESOURCE_TO_ES_MAPPING[resource_type].get(id=resource_es_id,
```

Comment:
instead of key->id->doc, can we just go key->doc

Reply:
skip the doc get, just doc update when getting the ID and the current value of the field